lib/runner/cli/cli.js

```diff
@@ -10,6 +12,52 @@ const ProcessListener = require('../process-listener.js');
 const analyticsCollector = require('../../utils/analytics.js');
 const {Logger, singleSourceFile} = Utils;
 
+class LocalTunnel {
+  constructor(settings = {}) {
+    this.settings = settings;
+    this.tunnel = this.settings.tunnel;
+    this.opts = this.tunnel ? this.tunnel.options : {};
+    this.key = this.opts.key;
+    if(this.tunnel && this.tunnel.type) {
+      switch(this.tunnel.type) {
+        case 'browserstack':
+          this.local = new BStackLocal();
+      }
+    }
+  }
+
+  async start() {
+    if (this.local && this.key) {
+      try {
+        await util.promisify(this.local.start.bind(this.local))({ key: this.key, ...this.opts });
+        this.localStarted = true;
+        this.settings.local_setup = {
+          localIdentifier: this.opts.localIdentifier
+        }
+        let signal = 0;
+        const handler = async () => {
+          signal++;
+          signal === 1 && await this.stop();
+          process.exit();
```

Comment:
This exits the parent nightwatch process, which is wrong. Ideally we want to only stop the tunnel. If possible can you use the interrupt handler to stop the tunnel. If that does not close the nightwatch process gracefully, then we might need to take a deeper look.